README.md

```diff
@@ -69,32 +69,103 @@ for example in data_loader:
     print(example)
 ```
 
+## Getting Started
 
-## Docker
+Currently Rikai is maintained for <a name="VersionMatrix"></a>Scala 2.12 and Python 3.7 and 3.8.
+
+There are multiple ways to install Rikai:
+
+1. Try it using the included [Dockerfile](#Docker).
+2. OR install it via pip `pip install rikai`, with 
+   [extras for aws/gc, pytorch/tf, and others](#Extras). 
+3. OR install it from [source](#Source)
+
+If you want to use Rikai with pyspark, please make sure you add the right jars to the [Spark
+options](#SparkSetup) at startup. Databricks users please see [setup in databricks](#Databricks) for
+details insteadf.
+
+### <a name="Docker"></a>Docker
 
 The included Dockerfile creates a standalone demo image with
 Jupyter, Pytorch, Spark, and rikai preinstalled with notebooks for you
 to play with the capabilities of the rikai feature store.
 
-### Building the image
+To build and run the docker image from the current directory:
+```bash
+# Clone the repo
+git clone git@github.com:eto-ai/rikai rikai
+# Build the docker image
+docker build --tag rikai --network host .
+# Run the image
+docker run -p 0.0.0.0:8888:8888/tcp rikai:latest jupyter lab -ip 0.0.0.0 --port 8888
+```
+
+If successful, the console should then print out a clickable link to JupyterLab. You can also
+open a browser tab and go to `localhost:8888`.
 
-1. First build the rikai jar
+### <a name="Extras"></a>Install from pypi
 
-``` bash
-# from rikai root
-sbt publishLocal # assumes Scala 2.12 and version is 0.0.1
-```
+Base rikai library can be installed with just `pip install rikai`. Dependencies for supporting
+pytorch (pytorch and torchvision), aws (boto), jupyter (matplotlib and jupyterlab) are all part of 
+optional extras. Many open-source datasets also use Youtube videos so we've also added pafy and 
+youtube-dl as optional extras as well. 
 
-``` bash
-# from rikai root
-# if the base system is linux and you want to expose the gpu, ensure `nvidia-docker2` is installed first
-docker build --tag rikai --network host .
-```
+For example, if you want to use pytorch in Jupyter to train models on rikai datasets in s3 
+containing Youtube videos you would run:
+
+`pip install rikai[pytorch,aws,jupyter,youtube]`
+
+If you're not sure what you need and don't mind installing some extra dependencies, you can 
+simply install everything:
 
-2. Running the image
+`pip install rikai[all]`
 
-``` bash
-docker run -p 0.0.0.0:8888:8888/tcp rikai:latest jupyter lab --ip 0.0.0.0 --port 8888 --NotebookApp.quit_button=True --NotebookApp.custom_display_url=http://127.0.0.1:8888
+### <a name="Source"></a>Install from source
+
+To build from source you'll need python as well as Scala with sbt installed:
+
+```bash
+# Clone the repo
+git clone git@github.com:eto-ai/rikai rikai
+# Build the jar
+sbt publishLocal
```

Comment:
i do have issue with this , to directly run with `pyspark --packages ai.eto.rikai:rikai_2.12:0.0.1` with missing netty deps. 

Lets make sure these processes getting verified automatically?

Reply:
```
java.io.FileNotFoundException: File file:/home/lei/.ivy2/jars/io.netty_netty-transport-native-epoll-4.1.46.Final.jar does not exist

	at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:611)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:824)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:601)
	at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:428)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1534)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1498)
	at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:494)
	at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:494)
	at scala.collection.immutable.List.foreach(List.scala:392)
```

Reply:
how did you setup Spark? With pre-packaged hadoop or user provided hadoop jars?